### DISCUSSION TITLE
Using rewrites with Plausible

### DISCUSSION THREAD

**Discussion by hkbertoson** [vercel/community#1082] [Help]
Hello,

I have created a vercel.json file that looks like this

````
{
	"rewrites": [
		{
			"source": "/stats/js/script.js",
			"destination": "https://plausible.io/js/script.js"
		},
		{
			"source": "/stats/api/event",
			"destination": "https://plausible.io/api/event"
		}
	]
}
````

Then I have added to my head. Deployed it but when loading the page after deployment. Its not working properly. Getting a 404 error for /stats/js/scripts.js
````
		<script
			defer
			src="/stats/js/script.js"
			data-domain="domain"
			data-api="/stats/api/event"
		></script>
````

**Comment by amyegan** [comment]
Thanks for reaching out @hkbertoson. You may get a `404` from https://plausible.io/api/event if things aren't set up correctly to post with the expected parameters. I would recommend double checking your `data-domain` setting. I assume you have it set to `"domain"` here to keep the actual value private, but you'll need that to make sure it matches your actual site domain.

If you're using Next.js, then you should use a slightly different setup to proxy Plausible: [plausible.io/docs/proxy/guides/nextjs](https://plausible.io/docs/proxy/guides/nextjs)

**Reply by hkbertoson** [reply]
I did double-check it. Here is the snippet from plausible.

I am using Astro. 

`<script defer data-domain="cityofmorrisok.gov" src="https://plausible.io/js/script.js"></script>`
Script from my code
````
		<script
			defer
			src="/stats/js/script.js"
			data-domain="cityofmorrisok.gov"
			data-api="/stats/api/event"
		></script>
````

**Reply by amyegan** [reply]
I tried to reproduce the problem by adding the rewrite config to a new [Astro Boilerplate](https://vercel.com/templates/astro/astro-boilerplate) project. I don't have a Plausible account set up, but the rewrite is working for me at [astro-boilerplate-xi.vercel.app/stats/js/script.js](https://astro-boilerplate-xi.vercel.app/stats/js/script.js).

I know a real-world app is going to be more complex than that simplified example. Could there be any other config properties affecting your redirects, either in your Astro app or in Vercel project settings that you have not already shared?
